numpy/core/src/npysort/quicksort.c.src

```diff
@@ -69,10 +90,12 @@ quicksort_@suff@(void *start, npy_intp num, void *NOT_USED)
     @type@ *stack[PYA_QS_STACK];
     @type@ **sptr = stack;
     @type@ *pm, *pi, *pj, *pk;
-    npy_intp depth_limit = npy_get_msb(num) * 2;
+    int depth[PYA_QS_STACK];
+    int * psdepth = depth;
+    int cdepth = NPY_SORT_MIN(PYA_QS_STACK, npy_get_msb(num) * 2);
 
     for (;;) {
-        if (depth_limit-- < 0) {
+        if (NPY_UNLIKELY(cdepth < 0)) {
```

Comment:
The pseudocode uses ` <= 0`, any reason for the change? Note that this would work fine if you didn't multiply by two above.